packages/core/upload/admin/src/components/TableList/CellContent.js

```diff
@@ -8,30 +8,13 @@ import { Typography } from '@strapi/design-system/Typography';
 import { PreviewCell } from './PreviewCell';
 import { formatBytes } from '../../utils';
 
-export const CellContent = ({
-  alternativeText,
-  content,
-  cellType,
-  elementType,
-  mime,
-  fileExtension,
-  thumbnailURL,
-  url,
-}) => {
+export const CellContent = ({ content, cellType, elementType, element }) => {
   const { formatDate, formatMessage } = useIntl();
 
   switch (cellType) {
     case 'image':
-      return (
-        <PreviewCell
-          alternativeText={alternativeText}
-          fileExtension={fileExtension}
-          mime={mime}
-          type={elementType}
-          thumbnailURL={thumbnailURL}
-          url={url}
-        />
-      );
+      return <PreviewCell type={elementType} element={element} />;
```

Comment:
I agree `element` is not clear - I struggled finding a name because it can be any kind of `media` but also a `folder` - wdyt? 👀

Reply:
`mediaOrFolderData` – clearest names are the best 😅

Reply:
What do you think about `cell`?

Reply:
mmh I think it could be confusing since `cellType` refers to the table column type (size, date, extension) while element can be image/video/file/folder and each can have size/date/extension - let me know if it's clear for you
that's why I differentiated `elementType` from `cellType` 🤔 @gu-stav 

same remarks for `mediaOrFolderData`, would that mean to rename `elementType` to `dataType`? @joshuaellis

Reply:
Mh. And why don't you just pass this as `content` like the others?

Reply:
content is not an object, it's the specific data for this column (e.g. date can be `updatedAt` or `createdAt`)
in the case of element here I only passed it because I needed the full element for `PreviewCell` which requires:
`const { alternativeText, ext, formats, mime, name, url } = element;`

first I passed them each as prop but thought it a bit messy since only one component (`PreviewCell`) will use them, but if you think it's better to pass each of them why not

Reply:
I don't think it would be a problem to pass `content` in different shapes. Content could be either a string or an object and depending on the switch it renders differently. I think I'd prefer that, because you can just pass around `content` and `CellContent` takes care of the formatting.

Reply:
Looks like you've sorted, I can go with `content`.

Reply:
wdyt about [this](https://github.com/strapi/strapi/pull/15274/commits/14a1565a0223ca23900ce29869b1b10982c6fe24#diff-f9064dd11a0df5b2757cab587ecd01a1444bf4ed1ad24c89a6639d673d21072e)? I find it clearer like that   👀
I'll update the tests to adapt the new changes if that works for you

Reply:
I think it looks good to me :)